### ISSUE TITLE
New release backwards compatible?

### ISSUE THREAD

**Issue by nmueller18** [RBoelter/mostViewed#8]
Thanks for the new release of mostViewed! However, is it supposed to be backwards compatible? On our system with OJS 3.1.2.4, the upgrade let to a "Failed AJAX request or invalid JSON returned" and complete outage. The fatal error in the php-log is obviously related to the change in the get-method from
```
$request = Application::getRequest();
```
to
```
$request = Application::get()->getRequest();
```
Below is the complete error log:
```
Sep  3 00:52:28 web08 php73-user: PHP Fatal error:  Uncaught Error: Call to undefined method Application::get() in /www/user/htdocs_ojs/plugins/generic/mostViewed/MostViewedPlugin.inc.php:35
Sep  3 00:52:28 web08 php73-user: Stack trace:
Sep  3 00:52:28 web08 php73-user: #0 /www/user/htdocs_ojs/lib/pkp/classes/plugins/PluginRegistry.inc.php(69): MostViewedPlugin->register('generic', 'plugins/generic...', NULL)
Sep  3 00:52:28 web08 php73-user: #1 /www/user/htdocs_ojs/lib/pkp/classes/plugins/PluginRegistry.inc.php(143): PluginRegistry::register('generic', Object(MostViewedPlugin), 'plugins/generic...', NULL)
Sep  3 00:52:28 web08 php73-user: #2 /www/user/htdocs_ojs/lib/pkp/classes/core/Dispatcher.inc.php(132): PluginRegistry::loadCategory('generic', true)
Sep  3 00:52:28 web08 php73-user: #3 /www/user/htdocs_ojs/lib/pkp/classes/core/PKPApplication.inc.php(252): Dispatcher->dispatch(Object(Request))
Sep  3 00:52:28 web08 php73-user: #4 /www/user/htdocs_ojs/index.php(68): PKPApplication->execute()
Sep  3 00:52:28 web08 php73-user: #5 {main}
Sep  3 00:52:28 web08 php73-user:  thrown in /www/user/htdocs_ojs/plugins/generic/mostViewed/MostViewedPlugin.inc.php on line 35
```

**Comment by RBoelter** [comment]
Damn, I just wanted to remove some deprecations, but yes this won't work in 3.1.
I'll fix this today.

**Comment by RBoelter** [comment]
I pushed the fix into a new stable-3_1_2 branch, because it is to complicated to handle deprecated code and support backwards compatibility in one plugin version. 
I wanted to create a release for this branch via `pkp-plugin bump`, but unfortunately it creates a release for the master branch even when stable-3_1_2 is checked out. Any sugessions @NateWr?

**Comment by RBoelter** [comment]
I created the tar file manually, you can find it here: https://github.com/RBoelter/mostViewed/releases/tag/3_1_2_v1_0_0-3

**Comment by nmueller18** [comment]
Many thanks for the quick fix! Now it works also with OJS 3.1.2.4! I will close the issue.

**Comment by NateWr** [comment]
> I wanted to create a release for this branch via pkp-plugin bump, but unfortunately it creates a release for the master branch even when stable-3_1_2 is checked out.

Hmm, I took a quick look through the CLI code but I didn't find anywhere that changes the branch. Can you run `pkp-plugin version` to see which version you have and check if you're on 0.6?
